#
# Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
#
getBuffer handling is tracked upstream as
http://www.icu-project.org/trac/ticket/12378
diff --git a/source/tools/genrb/reslist.cpp b/source/tools/genrb/reslist.cpp
--- a/icu/source/tools/genrb/reslist.cpp
+++ b/icu/source/tools/genrb/reslist.cpp
@@ -1458,8 +1458,16 @@ compareStringSuffixes(const void * /*con
     const StringResource *left = *((const StringResource **)l);
     const StringResource *right = *((const StringResource **)r);
     const UChar *lStart = left->getBuffer();
+    if ( lStart == NULL ) {
+        fprintf(stderr, "Could not allocate memory!!");
+        exit(U_MEMORY_ALLOCATION_ERROR);
+    }
     const UChar *lLimit = lStart + left->length();
     const UChar *rStart = right->getBuffer();
+    if ( rStart == NULL ) {
+        fprintf(stderr, "Could not allocate memory!!");
+        exit(U_MEMORY_ALLOCATION_ERROR);
+    }
     const UChar *rLimit = rStart + right->length();
     int32_t diff;
     /* compare keys in reverse character order */
